src/problems_mortar.jl

```diff
@@ -63,13 +65,202 @@ function get_formulation_type(problem::Problem{Mortar})
 end
 
 function assemble!(problem::Problem{Mortar}, time::Float64)
+    if length(problem.elements) == 0
+        warn("No elements defined in interface $(problem.name), this will result empty assembly!")
+        return
+    end
     if problem.properties.dimension == -1
         problem.properties.dimension = dim = size(first(problem.elements), 1)
-        info("assuming dimension of mesh tie surface is $dim")
-        info("if this is wrong set is manually using problem.properties.dimension")
+        info("Assuming dimension of mesh tie surface is $dim. If this is wrong set is manually using problem.properties.dimension")
     end
     dimension = Val{problem.properties.dimension}
     use_forwarddiff = Val{problem.properties.use_forwarddiff}
     assemble!(problem, time, dimension, use_forwarddiff)
 end
 
+""" Given a CCW ordered set of vertices, calculate area of polygon.
+
+Examples
+--------
+
+julia> P = Vector[[1/3, 5/12, 1/2], [1/3, 1/2, 1/2], [1/2, 1/2, 1/2], [1/2, 1/3, 1/2], [5/12, 1/3, 1/2]]
+5-element Array{Array{T,1},1}:
+ [0.333333,0.416667,0.5]
+ [0.333333,0.5,0.5]
+ [0.5,0.5,0.5]
+ [0.5,0.333333,0.5]
+ [0.416667,0.333333,0.5]
+
+julia> A = calculate_polygon_area(P)
+0.02430555555555556
+
+julia> isapprox(A, 7/288)
+true
+
+"""
+function calculate_polygon_area(P)
+    N_P = length(P)
+    A = sum([norm(1/2*cross(P[i]-P[1], P[mod(i,N_P)+1]-P[1])) for i=2:N_P])
+    return A
+end
+
+""" Function to print useful debug information from interface to find bugs. """
+function diagnose_interface(problem::Problem{Mortar}, time::Float64)
+    info("Diagnosing Mortar interface...")
+    props = problem.properties
+    field_dim = get_unknown_field_dimension(problem)
+    field_name = get_parent_field_name(problem)
+    slave_elements = get_slave_elements(problem)
+    
+    I_area = 0.0
+
+    if props.split_quadratic_slave_elements
+        info("props.split_quadratic_slave_elements = true")
+        if !props.linear_surface_elements
+            warn("Mortar3D: split_quadratic_surfaces = true and linear_surface_elements = false maybe have unexpected behavior")
+        end
+        slave_elements = split_quadratic_elements(slave_elements, time)
+    end
+    info("Number of slave elements in interface: $(length(slave_elements))")
+
+    # 1. calculate nodal normals and tangents for slave element nodes j ∈ S
+    normals = calculate_normals(slave_elements, time, Val{2};
+                                rotate_normals=props.rotate_normals)
+    update!(slave_elements, "normal", time => normals)
+
+    S_areas = []
+    C_areas = []
+    P_areas = []
+
+    for slave_element in slave_elements
+
+        info(repeat("-", 80))
+        info("Processing slave element $(slave_element.id), type = $(get_element_type(slave_element))")
+        info(repeat("-", 80))
```

Comment:
I think log level is correct in case of this function (``diagnose_interface``), because it's used as a debug tool for small test problems. Maybe the function itself should be somewhere else.